internal/pkg/describe/backend_service.go

```diff
@@ -10,60 +10,52 @@ import (
 	"sort"
 	"text/tabwriter"
 
-	"github.com/aws/copilot-cli/internal/pkg/config"
-	"github.com/aws/copilot-cli/internal/pkg/deploy"
 	"github.com/aws/copilot-cli/internal/pkg/deploy/cloudformation/stack"
+	"github.com/aws/copilot-cli/internal/pkg/manifest"
 	"github.com/aws/copilot-cli/internal/pkg/term/color"
 )
 
 // BackendServiceDescriber retrieves information about a backend service.
 type BackendServiceDescriber struct {
-	service         *config.Service
+	app             string
+	svc             string
 	enableResources bool
 
 	store                deployStoreSvc
-	svcDescriber         svcDescriber
+	svcDescriber         map[string]svcDescriber
 	initServiceDescriber func(string) error
 }
 
+// NewBackendServiceConfig contains fields that initiates BackendServiceDescriber struct.
+type NewBackendServiceConfig struct {
+	NewServiceConfig
+	EnableResources bool
+	DeployStore     deployStoreSvc
+}
+
 // NewBackendServiceDescriber instantiates a backend service describer.
-func NewBackendServiceDescriber(app, svc string) (*BackendServiceDescriber, error) {
-	configStore, err := config.NewStore()
-	if err != nil {
-		return nil, fmt.Errorf("connect to store: %w", err)
-	}
-	meta, err := configStore.GetService(app, svc)
-	if err != nil {
-		return nil, err
-	}
-	deployStore, err := deploy.NewStore(configStore)
-	if err != nil {
-		return nil, fmt.Errorf("connect to deploy store: %w", err)
+func NewBackendServiceDescriber(opt NewBackendServiceConfig) (*BackendServiceDescriber, error) {
+	describer := &BackendServiceDescriber{
+		app:             opt.App,
+		svc:             opt.Svc,
+		enableResources: opt.EnableResources,
+		store:           opt.DeployStore,
+		svcDescriber:    make(map[string]svcDescriber),
 	}
-	opts := &BackendServiceDescriber{
-		service:         meta,
-		enableResources: false,
-		store:           deployStore,
-	}
-	opts.initServiceDescriber = func(env string) error {
-		d, err := NewServiceDescriber(app, env, svc)
+	describer.initServiceDescriber = func(env string) error {
+		d, err := NewServiceDescriber(NewServiceConfig{
```

Comment:
what do you think of adding:
```suggestion
	describer.initServiceDescriber = func(env string) error {
		if d, ok := describer.svcDescriber[env]; ok {
			return d
		}
		d, err := NewServiceDescriber(NewServiceConfig{
```